tiledb/sm/filter/float_scaling_filter.cc

```diff
@@ -260,13 +261,34 @@ Status FloatScalingFilter::set_option_impl(
 
   switch (option) {
     case FilterOption::SCALE_FLOAT_BYTEWIDTH: {
-      byte_width_ = *(uint64_t*)value;
+      // The byte width parameter should be one of the sizes of the possible integer types. 
+      auto val = *(uint64_t*)value;
+      if (val != sizeof(int8_t) && val != sizeof(int16_t) 
+       && val != sizeof(int32_t) && val != sizeof(int64_t)) {
+        return LOG_STATUS(
+          Status_FilterError("Float scaling filter error; invalid byte width value."));
+      }
+      byte_width_ = val;
     } break;
     case FilterOption::SCALE_FLOAT_FACTOR: {
-      scale_ = *(double*)value;
+      // The scaling parameter should be not a NaN, infinity, or zero.
+      auto val = *(double*)value;
+      auto classify = std::fpclassify(val);
+      if (classify == FP_INFINITE || classify == FP_NAN || classify == FP_ZERO) {
```

Comment:
If we're going to check, we might want to throw FP_SUBNORMAL as another invalid option here.